### DISCUSSION TITLE
Building fails - unresolved external symbol linker errors

### DISCUSSION THREAD

**Discussion by holly564** [musescore/MuseScore#17358] [General]
Hello,

The folder "dependencies" has been added under Musescore directory (see picture)
![VerifyDependenciesLocation](https://user-images.githubusercontent.com/17574699/234361851-1633c4fb-5b3b-43c4-8fb5-e32586095e50.PNG)

But I get **linker errors**
> iex_braille_tests.vcxproj -> C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\importexport\braille\tests\RelWithDebInfo\iex_braille_tests.exe
     Creating library C:/Users/Holly/MuseScore/builds/Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo/src/app/RelWithDebInfo/MuseScore4.lib and object C:/Users/Holly/MuseScore/builds/Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo/src/app/RelWithDebInfo/MuseScore4.exp
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_GetRestrictedErrorInfo referenced in function "public: __cdecl winrt::hresult_error::hresult_error(struct winrt::hresult)" (??0hresult_error@winrt@@QEAA@Uhresult@1@@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_RoGetActivationFactory referenced in function "struct winrt::Windows::Devices::Enumeration::IDeviceInformationStatics __cdecl winrt::get_activation_factory<struct winrt::Windows::Devices::Enumeration::IDeviceInformationStatics>(struct winrt::param::hstring const &)" (??$get_activation_factory@UIDeviceInformationStatics@Enumeration@Devices@Windows@winrt@@@winrt@@YA?AUIDeviceInformationStatics@Enumeration@Devices@Windows@0@AEBUhstring@param@0@@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_RoOriginateLanguageException referenced in function "public: __cdecl winrt::hresult_error::hresult_error(struct winrt::hresult)" (??0hresult_error@winrt@@QEAA@Uhresult@1@@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_SetRestrictedErrorInfo referenced in function "public: struct winrt::hresult __cdecl winrt::hresult_error::to_abi(void)const " (?to_abi@hresult_error@winrt@@QEBA?AUhresult@2@XZ) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsCreateString referenced in function "private: virtual struct winrt::hstring __cdecl winrt::implements<struct winrt::impl::input_iterable<struct winrt::hstring,class std::vector<struct winrt::hstring,class std::allocator<struct winrt::hstring> > >,struct winrt::non_agile,struct winrt::no_weak_ref,struct winrt::Windows::Foundation::Collections::IIterable<struct winrt::hstring> >::GetRuntimeClassName(void)const " (?GetRuntimeClassName@?$implements@U?$input_iterable@Uhstring@winrt@@V?$vector@Uhstring@winrt@@V?$allocator@Uhstring@winrt@@@std@@@std@@@impl@winrt@@Unon_agile@3@Uno_weak_ref@3@U?$IIterable@Uhstring@winrt@@@Collections@Foundation@Windows@3@@winrt@@EEBA?AUhstring@2@XZ) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsCreateStringReference referenced in function "auto __cdecl winrt::impl::call_factory<struct winrt::Windows::Devices::Enumeration::DeviceInformation,struct winrt::Windows::Devices::Enumeration::IDeviceInformationStatics,class <lambda_7a3fc50f2aff298f3f6d4413e3038d19> >(class <lambda_7a3fc50f2aff298f3f6d4413e3038d19> &&)" (??$call_factory@UDeviceInformation@Enumeration@Devices@Windows@winrt@@UIDeviceInformationStatics@2345@V<lambda_7a3fc50f2aff298f3f6d4413e3038d19>@@@impl@winrt@@YA?A_P$$QEAV<lambda_7a3fc50f2aff298f3f6d4413e3038d19>@@@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsDuplicateString referenced in function "public: __cdecl winrt::param::async_iterable<struct winrt::hstring>::async_iterable<struct winrt::hstring>(class std::initializer_list<struct winrt::hstring>)" (??0?$async_iterable@Uhstring@winrt@@@param@winrt@@QEAA@V?$initializer_list@Uhstring@winrt@@@std@@@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsDeleteString referenced in function "public: __cdecl winrt::param::async_iterable<struct winrt::hstring>::async_iterable<struct winrt::hstring>(class std::initializer_list<struct winrt::hstring>)" (??0?$async_iterable@Uhstring@winrt@@@param@winrt@@QEAA@V?$initializer_list@Uhstring@winrt@@@std@@@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsPreallocateStringBuffer referenced in function "struct winrt::hstring __cdecl winrt::to_hstring<char const *,void>(char const * const &)" (??$to_hstring@PEBDX@winrt@@YA?AUhstring@0@AEBQEBD@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsDeleteStringBuffer referenced in function "public: __cdecl winrt::impl::hstring_builder::~hstring_builder(void)" (??1hstring_builder@impl@winrt@@QEAA@XZ) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsPromoteStringBuffer referenced in function "struct winrt::hstring __cdecl winrt::to_hstring<char const *,void>(char const * const &)" (??$to_hstring@PEBDX@winrt@@YA?AUhstring@0@AEBQEBD@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
audio.lib(unity_1_cxx.obj) : error LNK2019: unresolved external symbol WINRT_WindowsGetStringRawBuffer referenced in function "private: virtual long __cdecl mu::audio::AudioDevicesListener::OnDefaultDeviceChanged(enum __MIDL___MIDL_itf_mmdeviceapi_0000_0000_0001,enum __MIDL___MIDL_itf_mmdeviceapi_0000_0000_0002,wchar_t const *)" (?OnDefaultDeviceChanged@AudioDevicesListener@audio@mu@@EEAAJW4__MIDL___MIDL_itf_mmdeviceapi_0000_0000_0001@@W4__MIDL___MIDL_itf_mmdeviceapi_0000_0000_0002@@PEB_W@Z) [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\RelWithDebInfo\MuseScore4.exe : fatal error LNK1120: 12 unresolved externals [C:\Users\Holly\MuseScore\builds\Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo\src\app\mscore.vcxproj]
CMake Error at build.cmake:249 (message):
  Build step failed with status 1.  See output above for details.

These errors were preceded by this message about Qt 5 missing packages
-- The following REQUIRED packages have not been found:

 * Qt5

_What does it mean that the Qt 5 package was not found?_

While other Qt 5 packages were found
-- The following OPTIONAL packages have been found:

 * Qt5QmlModels (required version >= 5.15.2)
 * Python
 * Threads
 * Qt5Gui
 * Qt5Network
 * Qt5NetworkAuth
 * Qt5Qml
 * Qt5Quick
 * Qt5QuickControls2
 * Qt5QuickTemplates2
 * Qt5QuickWidgets
 * Qt5Xml
 * Qt5XmlPatterns
 * Qt5Svg
 * Qt5Widgets
 * Qt5PrintSupport
 * Qt5OpenGL
 * Qt5LinguistTools
 * Qt5Concurrent
 * Qt5WinExtras

-- The following REQUIRED packages have been found:

 * Qt5Core

What am I doing wrong? 
Thanks for any insight.

**Comment by Jojo-Schmitz** [comment]
Are you on Windows 10 or later? Which version exactly?

**Reply by holly564** [reply]
Windows 10 Pro 22H2
Plus SDK version 
![image](https://user-images.githubusercontent.com/17574699/234364503-5b226de6-c585-4ffc-8aef-b5a9bbf60a37.png)

**Reply by holly564** [reply]
Sorry, just found out that someone else had similar issues in https://github.com/musescore/MuseScore/issues/15361

**Reply by holly564** [reply]
Yeah!!!
![champagne](https://user-images.githubusercontent.com/17574699/234402341-da314ebb-8c52-4b3d-9e8f-9b15af3314a1.png)

-- Up-to-date: C:/Users/Holly/MuseScore/builds/Win-Qt5.15.2-msvc2019_64-VS16-RelWithDebInfo/install/bin

build.cmake: Complete after 59 minutes and 36 seconds

**Reply by Jojo-Schmitz** [reply]
Whow, that's long, takes less than 5 minutes for me
